lib/Sema/CSDiagnostics.cpp

```diff
@@ -9338,7 +9338,12 @@ bool InvalidMemberReferenceWithinInitAccessor::diagnoseAsError() {
 }
 
 bool ConcreteTypeSpecialization::diagnoseAsError() {
-  emitDiagnostic(diag::not_a_generic_type, ConcreteType);
+  emitDiagnostic(diag::not_a_generic_type, resolveType(ConcreteType));
```

Comment:
Nit: I'd prefer the resolveType happened during construction of the diagnostic, it's usually safer that way.

Reply:
Done. That was an accidental leftover from the changes along the way.